packages/pyolite-kernel/src/kernel.ts

```diff
@@ -192,23 +202,20 @@ export class PyoliteKernel extends BaseKernel implements IKernel {
   async executeRequest(
     content: KernelMessage.IExecuteRequestMsg['content']
   ): Promise<KernelMessage.IExecuteReplyMsg['content']> {
-    const result = await this._sendRequestMessageToWorker('execute-request', content);
-
-    return {
-      execution_count: this.executionCount,
-      ...result,
-    };
+    const result = await this._workerKernel.execute(content, this.parent);
+    result.execution_count = this.executionCount;
+    return result;
   }
 
   /**
    * Handle an complete_request message
    *
    * @param msg The parent message.
    */
-  async completeRequest(
+  completeRequest(
     content: KernelMessage.ICompleteRequestMsg['content']
   ): Promise<KernelMessage.ICompleteReplyMsg['content']> {
-    return await this._sendRequestMessageToWorker('complete-request', content);
+    return this._workerKernel.complete(content, this.parent);
```

Comment:
i personally like leaving the `await` in, even if it's just returning, as it will give more precise error stacks.

Reply:
Sure I'll add it back, thanks for the review.

I've been wondering if we could not even have the "workerKernel" be the one that implements the `IKernel` interface, because the interfaces become very similar now. This would need some restructure.